ethereum/blockwatch/block_watcher.go

```diff
@@ -139,10 +142,21 @@ func (w *Watcher) FastSyncToLatestBlock(ctx context.Context) (blocksElapsed int,
 			return blocksElapsed, err
 		}
 		if len(events) > 0 {
+			newMiniHeaders, err := w.stack.PeekAll()
+			if err != nil {
```

Comment:
I think that we should always handle errors rather than leaving them be if they are always `nil` (this protects us if an error is returned in the future). In this case, `PeekAll` should just not return an error.